src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -582,11 +582,13 @@ trait_ref ::= path_generic_args_without_colons
 // Impl
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 
-impl_item ::= UNSAFE? IMPL generic_params type where_clause? '{' inner_attr* impl_member* '}'
-            | UNSAFE? IMPL generic_params '!'? trait_ref FOR (type where_clause? '{' inner_attr* impl_member* '}' | '..' '{' '}') {
-  pin(".*") = "'{'"
+impl_item ::= UNSAFE? IMPL generic_params type where_clause? impl_members_block
+            | UNSAFE? IMPL generic_params '!'? trait_ref FOR (type where_clause? impl_members_block | '..' '{' '}') {
+  pin(".*") = '{'
 }
 
+impl_members_block ::= '{' (inner_attr | impl_member)* '}'
```

Comment:
This accepts more programs then the previous implementation and rust compiler. In particular,

```
impl S {
    fn foo() {}
    #![cfg(test)]
}
```

should be a syntax error.

Of course we can handle it on the semantic level, but I think that doing this in grammar is better.